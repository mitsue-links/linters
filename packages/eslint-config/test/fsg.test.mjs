import path from 'path';
import {lintFile} from './helper.mjs';

test('fsg_ok.mjs', async () => {
    const filePath = 'fsg_ok.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toStrictEqual([{
        filePath,
        messages: [],
    }]);
});

test('fsg_error.mjs', async () => {
    const filePath = 'fsg_error.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toMatchSnapshot();
});
