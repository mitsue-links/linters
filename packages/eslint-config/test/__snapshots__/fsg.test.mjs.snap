// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fsg_error.mjs 1`] = `
[
  {
    "filePath": "fsg_error.mjs",
    "messages": [
      {
        "line": 3,
        "ruleId": "no-undef",
        "severity": 2,
      },
      {
        "line": 6,
        "ruleId": "no-undef",
        "severity": 2,
      },
      {
        "line": 10,
        "ruleId": "one-var",
        "severity": 2,
      },
      {
        "line": 13,
        "ruleId": "no-undef",
        "severity": 2,
      },
      {
        "line": 13,
        "ruleId": "no-unused-expressions",
        "severity": 2,
      },
      {
        "line": 13,
        "ruleId": "no-sequences",
        "severity": 2,
      },
      {
        "line": 14,
        "ruleId": "no-undef",
        "severity": 2,
      },
      {
        "line": 14,
        "ruleId": "no-undef",
        "severity": 2,
      },
      {
        "line": 28,
        "ruleId": "camelcase",
        "severity": 2,
      },
      {
        "line": 41,
        "ruleId": "no-eval",
        "severity": 2,
      },
      {
        "line": 43,
        "ruleId": "no-new-func",
        "severity": 2,
      },
      {
        "line": 49,
        "ruleId": "no-unused-vars",
        "severity": 2,
      },
      {
        "line": 53,
        "ruleId": "no-implied-eval",
        "severity": 2,
      },
      {
        "line": 64,
        "ruleId": "no-caller",
        "severity": 2,
      },
      {
        "line": 71,
        "ruleId": "no-restricted-globals",
        "severity": 2,
      },
      {
        "line": 72,
        "ruleId": "no-restricted-globals",
        "severity": 2,
      },
      {
        "line": 74,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      {
        "line": 78,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      {
        "line": 79,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      {
        "line": 80,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      {
        "line": 82,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      {
        "line": 83,
        "ruleId": "no-restricted-properties",
        "severity": 2,
      },
      {
        "line": 89,
        "ruleId": "eqeqeq",
        "severity": 2,
      },
      {
        "line": 114,
        "ruleId": "one-var",
        "severity": 2,
      },
      {
        "line": 142,
        "ruleId": "curly",
        "severity": 2,
      },
      {
        "line": 146,
        "ruleId": "no-useless-assignment",
        "severity": 2,
      },
    ],
  },
]
`;
